app/helpers/application_helper.rb

```diff
@@ -265,6 +265,12 @@ def holding_block_search(document)
                                         class: 'availability-icon more-info', title: 'Click on the record for full availability info',
                                         'data-toggle' => 'tooltip').html_safe, class: 'empty', data: { record_id: document['id'] })
     end
+
+    if block.empty? && links.present?
+      # All other options came up empty but since we have electronic access let's show Online rather than no holdings.
+      block << content_tag(:span, 'Online', class: 'availability-icon badge badge-primary', title: 'Electronic access', 'data-toggle' => 'tooltip')
+    end
```

Comment:
I think this is ok for now. Can you add like: 
```
https://github.com/pulibrary/orangelight/blob/main/app/helpers/application_helper.rb#L222-L223
```
We can update lines https://github.com/pulibrary/orangelight/blob/main/app/helpers/application_helper.rb#L213-L224 later. I would need to look closer.

Reply:
Notice that the changes in this PR are **outside** of the holdings loop because there are no holdings (only links) for this record. 

The way I implement this fix I let all the previous logic take place and, if at the end of all the existing logic the (HTML) block is empty but there are links available then it adds the badge. I implement it this way since I did not want to mess with the existing logic and potentially break other holdings/links scenarios.

Reply:
https://github.com/pulibrary/orangelight/blob/9bde667d4f25588cd7a17e4dc996754233cbcb32/app/helpers/application_helper.rb#L426

Reply:
@christinach this was a fantastic suggestion. Reusing the code logic that you suggested displays both the Online badge and the text of the link (see screenshot above)

https://github.com/pulibrary/orangelight/blob/issue_2668_online_holdings/app/helpers/application_helper.rb#L269-L276

Reply:
Great @hectorcorrea !